apps/extensions_manager/lib/extensions_manager/model_extension.ex

```diff
@@ -0,0 +1,75 @@
+defmodule ExtensionsManager.ModelExtension do
+  defmacro __using__(_opts) do
+    quote do
+      Module.register_attribute(__MODULE__, :schema_changes, accumulate: true)
+      Module.register_attribute(__MODULE__, :method_block, accumulate: true)
+
+      import ExtensionsManager.ModelExtension, only: [add_to_schema: 4, add_to_schema: 3, include_method: 1]
+      @before_compile ExtensionsManager.ModelExtension
+
+      defmacro __using__(_opts) do
```

Comment:
Yes, Unfortunately this will prevent the another __using__/1 declaration in `ExtensionsManager.ExtendUser` (where its currently being used)

`ExtensionsManager.ExtendUser` is a special module wrapping up _schema changes_ and _support functions_ for `Nectar.User` (Ecto Model) from other modules (extensions) so we never envisaged a situation where another __using__ macro would be required for `ExtensionsManager.ExtendUser`

But for exceptional cases, we can probably add a `defoverridable [__using__: 1]` in `ExtensionsManager.ExtendUser` trusting the developer's intent for change :thought_balloon: :guardsman: